tests/testTableUtils.py

```diff
@@ -0,0 +1,181 @@
+    @param[in] maxPix  maximum pixel position; used for both x and y;
+                        the min is the negative of maxPix
+    @param[in] numPoints  number of points in x or y; total points = numPoints*numPoints
+    """
+    if len(cat) != 0:
+        raise RuntimeError("cat must be empty")
+    for i in np.linspace(-maxPix, maxPix, numPoints):
+        for j in np.linspace(-maxPix, maxPix, numPoints):
+            rec = cat.addNew()
+            rec.set(key, afwGeom.Point2D(i, j))
+
+
+class UpdateTestCase(lsst.utils.tests.TestCase):
+    """A test case for the lsst.afw.table.updateRefCentroids and updateSourceCoords
+    """
+
+    def setUp(self):
+        self.crval = afwCoord.IcrsCoord(afwGeom.PointD(44., 45.))
+        self.crpix = afwGeom.Point2D(15000, 4000)
+
+        arcsecPerPixel = 1/3600.0
+        CD11 = arcsecPerPixel
+        CD12 = 0
+        CD21 = 0
+        CD22 = arcsecPerPixel
+
+        self.wcs = afwImage.makeWcs(self.crval, self.crpix, CD11, CD12, CD21, CD22)
+
+        refSchema = afwTable.SimpleTable.makeMinimalSchema()
+        self.refCentroidKey = afwTable.Point2DKey.addFields(refSchema, "centroid", "centroid", "pixels")
+        self.refCoordKey = afwTable.CoordKey(refSchema["coord"])
+        self.refCat = afwTable.SimpleCatalog(refSchema)
+
+        # an alias is required to make src.getCentroid() work;
+        # simply defining a field named "slot_Centroid" doesn't suffice
+        srcSchema = afwTable.SourceTable.makeMinimalSchema()
+        self.srcCentroidKey = afwTable.Point2DKey.addFields(srcSchema, "base_SdssCentroid",
+                                                            "centroid", "pixels")
+        srcAliases = srcSchema.getAliasMap()
+        srcAliases.set("slot_Centroid", "base_SdssCentroid")
+        self.srcCoordKey = afwTable.CoordKey(srcSchema["coord"])
+        self.sourceCat = afwTable.SourceCatalog(srcSchema)
+
+    def tearDown(self):
+        del self.wcs
+        del self.refCat
+        del self.sourceCat
+
+    def testNull(self):
+        """Check that an empty list causes no problems for either function"""
+        afwTable.updateRefCentroids(self.wcs, [])
+        afwTable.updateSourceCoords(self.wcs, [])
+
+    def testRefCenter(self):
+        """Check that a ref obj at the center is handled as expected"""
+        refObj = self.refCat.addNew()
+        refObj.set(self.refCoordKey, self.crval)
+
+        # initial centroid should be nan
+        nanRefCentroid = self.refCat[0].get(self.refCentroidKey)
+        for val in nanRefCentroid:
+            self.assertTrue(math.isnan(val))
+
+        # computed centroid should be crpix
+        afwTable.updateRefCentroids(self.wcs, self.refCat)
+        refCentroid = self.refCat[0].get(self.refCentroidKey)
+        self.assertPairsNearlyEqual(refCentroid, self.crpix)
+
+        # coord should not be changed
+        self.assertPairsNearlyEqual(self.refCat[0].get(self.refCoordKey), self.crval, maxDiff=0)
```

Comment:
Ah, right, it's an afw object. Can you try it with equal and see what happens? I see "nearlyEqual" and think "something could have gotten jittered a bit", even with the maxDiff.